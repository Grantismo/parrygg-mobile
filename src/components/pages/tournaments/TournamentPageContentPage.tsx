import React, { useRef, useState } from "react";
import { FieldErrors, RegisterOptions, useForm } from "react-hook-form";
import { GestureResponderEvent, TouchableOpacity, View } from "react-native";
import DraggableFlatList, {
  RenderItemParams,
  ScaleDecorator,
  ShadowDecorator,
} from "react-native-draggable-flatlist";

import Background from "@/components/base/Background";
import Button from "@/components/base/Button";
import CollapsableCard from "@/components/base/CollapsableCard";
import Text from "@/components/base/Text";
import Nav from "@/components/base/navigation/Nav";
import Form from "@/components/forms/Form";
import Input from "@/components/forms/Input";
import tw from "@/lib/tailwind";
import DownArrow from "@assets/icons/DownArrow";
import DragHandle from "@assets/icons/DragHandle";
import Plus from "@assets/icons/Plus";
import RightArrow from "@assets/icons/RightArrow";

interface Props {
  title: string;
  dragProps: {
    drag: (event: GestureResponderEvent) => void;
    disabled: boolean;
  };
}

type PageContentSectionData = {
  tag: string;
  firstName: string;
  lastName: string;
  bio: string;
  pronouns: string;
};

const PageContentSectionForm = ({
  title,
  dragProps: { drag, disabled },
}: Props) => {
  const errors: FieldErrors<PageContentSectionData> = {};

  const validation: RegisterOptions[] = [];

  const { handleSubmit, register, setValue } = useForm<FormData>({ errors });
  const ref = useRef<{ setOpen: (open: boolean) => void }>(null);

  const onSubmit = (data: FormData) => {
    ref.current?.setOpen(false);
  };

  return (
    <CollapsableCard
      ref={ref}
      style={tw`w-full px-5 pt-3.5 pb-4.5 rounded-lg`}
      header={(open) => {
        return (
          <View style={tw`w-full flex flex-row items-center justify-between`}>
            <View style={tw`flex-row items-center justify-center`}>
              <TouchableOpacity
                hitSlop={10}
                onPressIn={drag}
                disabled={disabled}
              >
                <DragHandle style={tw`mr-2`} color="white" />
              </TouchableOpacity>
              <Text>{title}</Text>
            </View>
            {!open && <RightArrow color="#FFC93F" />}
            {open && <DownArrow color="white" />}
          </View>
        );
      }}
    >
      <View style={tw`pt-6`}>
        <Form<FormData> {...{ register, setValue, validation, errors }}>
          <Input
            value={title}
            name="sectionTitle"
            label="Section Title"
            placeholder="Section Title"
            required
          />
          <Input
            name="sectionContent"
            label="Section Content"
            placeholder="Text or markup"
            numberOfLines={8}
            textAlignVertical="top"
            multiline
            required
          />
          <View style={tw`flex-row`}>
            <Button
              size="sm"
              color="red"
              title="Delete"
              style={tw`flex-1 mr-2`}
              pressableStyle={tw`py-2`}
            />
            <Button
              size="sm"
              title="Save"
              style={tw`flex-1`}
              pressableStyle={tw`py-2`}
              onPress={handleSubmit(onSubmit)}
            />
          </View>
        </Form>
      </View>
    </CollapsableCard>
  );
};

const TournamentPageContentPage = () => {
  const [sections, setSections] = useState([
    "First Section",
    "Why attend Genesis",
  ]);

  const renderItem = ({ item, drag, isActive }: RenderItemParams<string>) => {
    return (
      <ShadowDecorator>
        <PageContentSectionForm
          title={item}
          dragProps={{ drag, disabled: isActive }}
        />
      </ShadowDecorator>
    );
  };

  return (
    <Background>
      <Nav title="Page Content" showBack />
      <View style={tw`w-full h-full`}>
        <DraggableFlatList
          data={sections}
          onDragEnd={({ data }) => setSections(data)}
          keyExtractor={(s) => s}
          renderItem={renderItem}
        />
        <Button style={tw`w-full`}>
          <Plus color="#1B1B1B" style={tw`mr-2`} />
          <Text color="black">Add Section</Text>
        </Button>
      </View>
    </Background>
  );
};

export default TournamentPageContentPage;
