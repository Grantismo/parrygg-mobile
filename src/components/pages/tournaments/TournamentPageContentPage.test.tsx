import { describe, expect, it } from "@jest/globals";
import React from "react";
import renderer from "react-test-renderer";

import TournamentPageContentPage from "@/components/pages/tournaments/TournamentPageContentPage";

describe("TournamentPageContentPage", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TournamentPageContentPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
