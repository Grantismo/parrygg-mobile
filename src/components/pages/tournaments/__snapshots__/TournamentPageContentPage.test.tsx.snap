// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TournamentPageContentPage renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#1b1b1b",
    }
  }
>
  <View
    style={
      {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "opacity": 0.03,
          "resizeMode": "cover",
        }
      }
    />
  </View>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      [
        {
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
        },
      ]
    }
  >
    <View>
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(10, 10, 10, 0)",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 20,
            "paddingLeft": 32,
            "paddingRight": 32,
            "paddingTop": 40,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "width": 56,
            }
          }
        />
        <Text
          collapsable={false}
          style={
            {
              "color": "#fff",
              "fontFamily": "Inter_600SemiBold",
              "fontSize": 20,
            }
          }
        >
          Page Content
        </Text>
        <View
          style={
            {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "width": 56,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            hitSlop={5}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Inter_500Medium",
                  },
                  {
                    "fontSize": 16,
                  },
                  {
                    "color": "#777",
                  },
                  undefined,
                ]
              }
            >
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "height": "100%",
              "paddingLeft": 24,
              "paddingRight": 24,
            },
            [
              {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 16,
                },
                {
                  "width": "100%",
                },
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              colors={
                [
                  4293899034,
                  4294953798,
                ]
              }
              style={
                [
                  {
                    "backgroundColor": "rgba(56, 52, 42, 0.38)",
                  },
                  {
                    "borderRadius": 10,
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderWidth": 2,
                      "color": "#000",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 15,
                      "paddingLeft": 24,
                      "paddingRight": 24,
                      "paddingTop": 13,
                    },
                    {
                      "borderColor": "#ffd978",
                    },
                    false,
                    false,
                    false,
                    false,
                    {
                      "borderRadius": 10,
                    },
                    undefined,
                  ]
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="18"
                  bbWidth="18"
                  fill="none"
                  focusable={false}
                  height={18}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "marginRight": 8,
                      },
                      {
                        "flex": 0,
                        "height": 18,
                        "width": 18,
                      },
                    ]
                  }
                  vbHeight={18}
                  vbWidth={18}
                  width={18}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M8.99809 0.0605469C4.06081 0.0605469 0.0585938 4.06276 0.0585938 9.00005C0.0585938 13.9373 4.06081 17.9395 8.99809 17.9395C13.9354 17.9395 17.9376 13.9373 17.9376 9.00005C17.9376 4.06276 13.9354 0.0605469 8.99809 0.0605469ZM13.4678 9.894H9.89204V13.4698H8.10414V9.894H4.52834V8.1061H8.10414V4.5303H9.89204V8.1061H13.4678V9.894Z"
                      fill={
                        {
                          "payload": 4279966491,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Inter_500Medium",
                      },
                      {
                        "fontSize": 16,
                      },
                      {
                        "color": "#000",
                      },
                      undefined,
                    ]
                  }
                >
                  Add Section
                </Text>
              </View>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
